piccolo/apps/migrations/auto/migration_manager.py

```diff
@@ -479,6 +479,35 @@ async def _run_alter_columns(self, backwards: bool = False):
                 if (old_column_class is not None) and (
                     column_class is not None
                 ):
+                    # Array columns migrations with base column type changes
+                    if (
+                        column_class.value_type == list
+                        and old_column_class.value_type == list
+                    ):
+                        old_array_column = old_column_class(**old_params)
+                        old_array_column._meta._table = _Table
+                        old_array_column._meta._name = alter_column.column_name
+                        old_array_column._meta.db_column_name = (
+                            alter_column.db_column_name
+                        )
+
+                        new_array_column = column_class(**params)
+                        new_array_column._meta._table = _Table
+                        new_array_column._meta._name = alter_column.column_name
+                        new_array_column._meta.db_column_name = (
+                            alter_column.db_column_name
+                        )
+                        # first drop array column
+                        await self._run_query(
+                            _Table.alter().drop_column(old_array_column)
+                        )
+                        # than add array column with new base column type
+                        await self._run_query(
+                            _Table.alter().add_column(
+                                name=new_array_column._meta.name,
+                                column=new_array_column,
+                            )
+                        )
                     if old_column_class != column_class:
```

Comment:
Rather than the logic you've added above, can't you just change this `if` statement to something like:

```python
base_colum_class = params.get('base_column').__class__
old_base_column_class = old_params.get('base_column').__class__

if (old_column_class != column_class) or (old_base_colum_class != base_colum_class):
    ...
```

Reply:
Great. Thanks. That's much better.